### PR TITLE
Process schedule as a circular queue

### PR DESCRIPTION
**Pull request by SuhasHebbar**
The schedule will now be treated as a circular queue. That is after a first pass over the schedule a second pass is carried out till the first connection whole departure is less than the arrival time of the itinerary.

### CODE REVIEW

**Comment by SuhasHebbar** [comment]
The latest commit 913a9ec5dfc631ca403aab6de888e8f4de4cde4f doesn't incorporate any new feature. It mainly involves converting hard tabs to soft tabs and running the elixir code formatter to better fit the standard coding style.

File: lib/ts/station/fsm.ex [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    new()
+    |> input_data(input)
+  end
+
+  def update_station(station_fsm, new_station_struct) do
+    station_fsm
+    |> update(new_station_struct)
+  end
+
+  def process_itinerary(station_fsm, itinerary) do
```
Comment (prasadtalasila) [inline lib/ts/station/fsm.ex]: please remind me the bug (a missing module API call) we found during the code review and your fix for the same.

File: lib/ts/station/fsm_behaviour.ex [hunk 2]
```diff
@@ -0,0 +1,13 @@
+defmodule Station.FsmBehaviour do
```
Comment (prasadtalasila) [inline lib/ts/station/fsm_behaviour.ex]: Station.FSMBehavior

File: lib/ts/station/fsm.ex [hunk 3]
```diff
@@ -0,0 +1,205 @@
+defmodule Station.Fsm do
```
[the remark this answers is not in the document]
Comment (SuhasHebbar) [reply lib/ts/station/fsm.ex]: That is doable, then the module will be named f_s_m.ex to keep to the naming conventions.

File: lib/ts/station/fsm_behaviour.ex (hunk 2 above)
Comment (SuhasHebbar) [reply lib/ts/station/fsm_behaviour.ex]: Similar to Station.FSM the module will then be named f_s_m_behaviour.

Comment (prasadtalasila) [reply lib/ts/station/fsm_behaviour.ex]: well. FSM is an abbreviation - not a regular work. As per camel case convention, we are allowed to keep all capitals even in the file name.

Comment (SuhasHebbar) [reply lib/ts/station/fsm_behaviour.ex]: Ok. then we'll have FSM.ex and FSM_behaviour.ex.

File: .travis.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 language: elixir
 elixir:
-- 1.5.3
+- 1.6.2
```
[the remark this answers is not in the document]
Comment (SuhasHebbar) [reply .travis.yml]: The versions have been bumped up.

File: lib/ts/util/itinerary.ex [hunk 5]
```diff
@@ -1,179 +1,243 @@
...
-		should_stop = stop_fn(neighbour_map, schedule) # Find out if stop or not
-
-		if should_stop == :true do
-			List.delete_at(vars_tail, 0)
-		else
-			find_valid_connection(vars)
-		end
-
-	end
+  @moduledoc """
```
[the remark this answers is not in the document]
Comment (SuhasHebbar) [reply lib/ts/util/itinerary.ex]: The itinerary now follows Util.ItineraryBehaviour.

**Comment by prasadtalasila** [comment]
@SuhasHebbar Can this PR be merged?

**Comment by SuhasHebbar** [comment]
Yes. The requested changes have been incorporated.

**Pull request merged**